src/components/RegisterForm/RegisterForm.jsx

```diff
@@ -1,99 +1,329 @@
+        id: "#confirmPassword",
+        errorMessage: "Password does not match"
+      }
+    },
+    {
+      condition: checkValidity(false, validPasswordFormat, "password"),
+      toolTipInfo: {
+        open: !validPasswordFormat,
+        id: "#password",
+        errorMessage:
+          "Invalid password (Must contain at least 1 upper case, lower case, number and symbol, and be between 8-20 characters."
+      }
+    },
+    {
+      condition: checkValidity(false, validFirstName, "firstName"),
+      toolTipInfo: {
+        open: !validFirstName,
+        id: "#firstName",
+        errorMessage: "Invalid First Name, no numbers or special characters."
+      }
+    },
+    {
+      condition: checkValidity(false, validMiddleName, "middleName"),
+      toolTipInfo: {
+        open: !validMiddleName,
+        id: "#middleName",
+        errorMessage: "Invalid Middle Name, no numbers or special characters."
+      }
+    },
+    {
+      condition: checkValidity(false, validLastName, "lastName"),
+      toolTipInfo: {
+        open: !validLastName,
+        id: "#lastName",
+        errorMessage: "Invalid Last Name, no numbers or special characters."
+      }
+    },
+    {
+      condition: checkValidity(false, validDOB, "birthDate"),
+      toolTipInfo: {
+        open: !validDOB,
+        id: "#birthDate",
+        errorMessage: "Can't be born in a future date."
+      }
+    },
+    {
+      condition: checkValidity(false, validPhone, "phoneNumber"),
+      toolTipInfo: {
+        open: !validPhone,
+        id: "#phoneNumber",
+        errorMessage:
+          "Must be 10 digit numbers (xxx) xxx xxxx (can be seperated by - or space)."
+      }
+    },
+    {
+      condition: checkValidity(false, validEmail, "email"),
+      toolTipInfo: {
+        open: !validEmail,
+        id: "#email",
+        errorMessage: "Invalid Email Format."
+      }
     }
-  
- 
+  ];
 
-    return (
+  return (
     <div className="regForm">
-      <RegisterForm handleButtonClick={handleButtonClick} onChange={onChange}/>
-      {/* find a way to map this!!!!  */}
-      { (validPassword === false && user_credentials.password !== "") && 
-            <Tooltip
-                open={!validPassword }
-                target="#confirm_password">
-            <span id="input_error"> Password does not match</span>
-            </Tooltip> ||
-        validPasswordFormat === false && user_credentials.password !== "" && 
+      <Alert
+        dismissible={triggerAlert}
+        open={alertNotification.isOpen}
+        className="mb-3"
+        theme={alertNotification.alertType}
+      >
+        {alertNotification.alertMessage}
+      </Alert>
+
+      <RegisterForm
+        handleButtonClick={handleButtonClick}
+        onChange={onChange}
+        validInputs={validInputs}
+        userCredentials={userCredentials}
+        validInvalidByName={validInvalidByName}
+      />
+      {conditions.map(
+        condition =>
+          condition.condition && (
```

Comment:
i think your checker `condition.condition` is redundant as there is no ambiguity because you manually defined the `condition` array.

Reply:
This is used to not display error message when input is erased.